pkg/agent/secondarynetwork/podwatch/controller_test.go

```diff
@@ -65,26 +72,40 @@ const (
 }`
 
 	defaultMTU    = 1500
+	defaultVLAN   = 100
 	sriovDeviceID = "sriov-device-id"
-
 	podName       = "pod1"
 	containerID   = "container1"
 	podIP         = "1.2.3.4"
 	networkName   = "net"
 	interfaceName = "eth2"
+	ovsPortUUID   = "12345678-e29b-41d4-a716-446655440000"
 )
 
-func testNetwork(name string) *netdefv1.NetworkAttachmentDefinition {
+func testNetwork(name, networkType string) *netdefv1.NetworkAttachmentDefinition {
+	configStr := strings.Replace(netAttachConfig, "network-type", networkType, 1)
 	return &netdefv1.NetworkAttachmentDefinition{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: name,
 		},
 		Spec: netdefv1.NetworkAttachmentDefinitionSpec{
-			Config: netAttachConfig,
+			Config: configStr,
 		},
 	}
 }
 
+func testNetworkExt(name, cniType, networkType string, mtu int, vlan uint16) *netdefv1.NetworkAttachmentDefinition {
+	def := testNetwork(name, networkType)
+	configStr := def.Spec.Config
+	if cniType != "" {
+		configStr = strings.Replace(configStr, "type\": \"antrea", "type\": \""+cniType, 1)
+	}
+	configStr = strings.Replace(configStr, "mtu\": 1500", "mtu\": "+strconv.Itoa(mtu), 1)
+	configStr = strings.Replace(configStr, "vlan\": 100", "vlan\": "+strconv.Itoa(int(vlan)), 1)
```

Comment:
IMO, you should use the `text/template` package to do this more cleanly

Reply:
Changed to `text/template`. Thanks for the suggestion.